### DISCUSSION TITLE
SASL Login Name Rejected: "You Are Sending Too Many Emails Too Fast" Error in Postfix

### DISCUSSION THREAD

**Discussion by Stefpls** [Mailu/Mailu#3496] [User support]
Hello everyone,

I'm running a Postfix mail server and encountering an issue where outgoing emails from a specific account are being rejected with the following error message: SASL login name rejected: You are sending too many emails too fast.; from=<mymail@mydomain.com>
My server is actually sending emails very actively from two accounts and this problem is related to these accounts.
The issue occurs when sending a high volume of emails, and the server rejects further attempts from this account, saying the emails are being sent too quickly. I'm unsure if this is a Postfix rate limiting configuration, SASL-related, or if something else is causing the problem.

SASL authentication enabled
Running in a Docker container (Mailu environment)
Any guidance or advice would be greatly appreciated!

Thank you!

**Comment by nextgens** [comment]
check ``MESSAGE_RATELIMIT`` and ``MESSAGE_RATELIMIT_EXEMPTION`` out on https://mailu.io/2024.06/configuration.html#mail-settings

**Reply by Stefpls** [reply]
Hello!

@nextgens, Thanks for the advice. I check MESSAGE_RATELIMIT and MESSAGE_RATELIMIT_EXEMPTION in the Mailu documentation here to resolve the rate limiting issue. I tried setting MESSAGE_RATELIMIT to 10000, but then the server stopped sending messages and produced errors like:

Sep 17 12:11:03 mail postfix/smtpd[1525]: warning: socketmap:unix:/tmp/podop.socket:senderrate socketmap server temporary error: unknown error
Sep 17 12:11:03 mail postfix/smtpd[1525]: warning: socketmap:unix:/tmp/podop.socket:senderrate lookup error for "mymail@mydomain.com"
Sep 17 12:11:03 mail postfix/smtpd[1525]: NOQUEUE: reject: RCPT from service.mydomain.com[172.30.30.4]: 451 4.3.5 <mymail@mydomain.com>: SASL login name rejected: Server configuration error; from=<mymail@mydomain.com> to=<cheftournant.mail@gmail.com> proto=ESMTP helo=<localhost>
ERROR:root:Error when processing request
Traceback (most recent call last):
  File "/app/venv/lib/python3.11/site-packages/podop/postfix.py", line 104, in process_request
    result = await table.get(key)
  File "/app/venv/lib/python3.11/site-packages/podop/table.py", line 35, in get
    raise Exception(request.status)
Exception: 500
It seems that setting such a high value caused a server configuration error within Mailu's rate-limiting mechanism.

To temporarily solve the initial problem, I used a script inside the container with the following postconf commands:

postconf -e 'queue_run_delay = 200s'
postconf -e 'smtp_destination_rate_delay = 1s'
postconf -e 'smtpd_client_message_rate_limit = 100'
postconf -e 'maximal_queue_lifetime = 12500s'
postconf -e 'bounce_queue_lifetime = 12500s'
postconf -e 'minimal_backoff_time = 300s'
postconf -e 'maximal_backoff_time = 4000s'
Now, everything is working fine. However, I'd like to automate this process so that these settings persist and apply automatically, even after container restarts or updates.

My questions are:

How can I properly configure MESSAGE_RATELIMIT and MESSAGE_RATELIMIT_EXEMPTION in Mailu without causing server errors?

Is there a recommended value for MESSAGE_RATELIMIT that avoids overwhelming the rate-limiting mechanism?
How can I use MESSAGE_RATELIMIT_EXEMPTION to exempt certain accounts from rate limiting?
What's the best way to automate the Postfix configurations within the Mailu environment?

Should I modify the Mailu configuration files or use environment variables to make these settings persistent?
Is there a way to integrate my postconf adjustments into the Mailu setup so that they are applied automatically on container startup?
Any guidance or advice would be greatly appreciated!

Thank you in advance for your help!

**Reply by ghostwheel42** [reply]
A limit consists of a number and a time-period like "100/day" or "50/minute".
So you probably wanted to sepcify "10000/day".
The parsing error should be catched though.

**Reply by Stefpls** [reply]
"Dear all,

I have successfully set the MESSAGE_RATELIMIT; however, the main issue now lies in automating the application of the previously mentioned postconf parameters.

Thank you for your attention to this matter."

This version is polite and conveys the message clearly.
